dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,325 @@
+import math
+import numpy as np
+import toolz
+from sklearn.base import clone
+from sklearn.model_selection import ParameterSampler
+from sklearn.utils import check_random_state
+
+from tornado import gen
+from toolz import groupby
+from dask.distributed import as_completed, default_client, futures_of
+from dask_searchcv.model_selection import DaskBaseSearchCV
+import dask.array as da
+from ..datasets import make_classification
+from ..wrappers import Incremental
+from ._split import train_test_split
+from distributed.metrics import time
+from distributed import Future
+from copy import deepcopy
+
+
+def _get_nr(R, eta=3):
+    s_max = math.floor(math.log(R, eta))
+    B = (s_max + 1) * R
+
+    brackets = list(reversed(range(int(s_max + 1))))
+    N = [math.ceil(B / R * eta**s / (s + 1)) for s in brackets]
+    R = [int(R * eta**-s) for s in brackets]
+    return list(map(int, N)), list(map(int, R)), brackets
+
+
+def _partial_fit(model_and_meta, X, y, meta=None, fit_params={},
+                 dry_run=False, **kwargs):
+    start = time()
+    assert isinstance(X, np.ndarray)
+    assert y is None or isinstance(y, np.ndarray)
+    model = deepcopy(model_and_meta[0])
+    if meta is None:
+        meta = deepcopy(model_and_meta[1])
+    else:
+        meta = deepcopy(meta)
+    meta['mean_copy_time'] += time() - start
+    while meta['iterations'] < meta['partial_fit_calls']:
+        if not dry_run:
+            model.partial_fit(X, y, **fit_params)
+        meta["iterations"] += 1
+    meta['mean_fit_time'] += time() - start
+    return model, meta
+
+
+def _score(model_and_meta, x, y, dry_run=False):
+    start = time()
+    model = deepcopy(model_and_meta[0])
+    meta = deepcopy(model_and_meta[1])
+    meta['mean_copy_time'] += time() - start
+    if dry_run:
+        score = 0
+        #  BUG: uncommenting these lines gives a bug in test_info for R=27, 81
+        #  seed = hash(((k, v) for k, v in meta.items())) % 2**32
+        #  rng = np.random.RandomState(seed)
+        #  score = rng.rand()
+    else:
+        # TODO: change this to dask_ml.get_scorer
+        score = model.score(x, y)
+    meta.update(score=score)
+    assert meta['iterations'] > 0
+    meta['mean_score_time'] += time() - start
+    meta['mean_test_score'] = score
+    return meta
+
+
+def _top_k(results, k=1, sort="score"):
+    res = toolz.topk(k, results, key=sort)
+    return list(res)
```

Comment:
I recommend that we just inline this function.  I believe that reducing the depth of calls in a function this will help reduce debugging time for others in the future.